pilot/pkg/model/service.go

```diff
@@ -119,11 +119,25 @@ const (
 )
 
 const (
-	// MTLSReadyLabelShortname name used for determining endpoint level tls transport socket configuration
-	MTLSReadyLabelShortname = "mtlsReady"
+	// TLSModeLabelShortname name used for determining endpoint level tls transport socket configuration
+	TLSModeLabelShortname = "tlsMode"
 
-	// MTLSReadyLabelName name for the mtlsReady label given to service instances to toggle mTLS autopilot
-	MTLSReadyLabelName = "security.istio.io/" + MTLSReadyLabelShortname
+	// TLSModeLabelName name for the mtlsReady label given to service instances to toggle mTLS autopilot
+	TLSModeLabelName = "security.istio.io/" + TLSModeLabelShortname
+)
+
+// TLSModeLabelValue indicates the TLS mode associated with the endpoint based on which the
+// transport socket will bet setup in the cluster
+type TLSModeLabelValue string
+
+const (
+	// DisabledTLSModeLabel implies that this endpoint should receive traffic as is (mostly plaintext)
+	DisabledTLSModeLabel TLSModeLabelValue = "disabled"
+	// SimpleNoVerifyTLSModeLabel implies that this endpoint should receive one-way TLS connection with
+	// root cert verification
+	SimpleNoVerifyTLSModeLabel TLSModeLabelValue = "simple-no-verify"
+	// IstioMutualTLSModeLabel implies that the endpoint is ready to receive Istio mTLS connections.
+	IstioMutualTLSModeLabel TLSModeLabelValue = "istio"
```

Comment:
I think `istio_mutual` is better than `istio`. mtls can not be assumed, if we're introducing more and more tls config mode.

Reply:
We don’t want a long label for each endpoint do we ?

Reply:
sure... good point.